src/modules/SX1261.cpp

```diff
@@ -0,0 +1,46 @@
+#include "SX1261.h"
+
+// note that this is untested (lacking the hardware) and based purely on the datasheet.
+int16_t SX1261::setOutputPower(int8_t power) {  
+  // check allowed power range
+  if (!((power >= -17) && (power <= 14))) {
+    return(ERR_INVALID_OUTPUT_POWER);
+  }
+
+  // get current OCP configuration
+  uint8_t ocp = 0;
+  int16_t state = readRegister(SX126X_REG_OCP_CONFIGURATION, &ocp, 1);
+  if (state != ERR_NONE) {
+    return(state);
+  }
+
+  state = setOptimalLowPowerPaConfig(&power);
```

Comment:
I think we wish to restore `OCP` even if `state != ERR_NONE`? The `setTxParams` is guarded by a `if (state == ERR_NONE)` so it doesn't get executed if `setOptimalLowPowerConfig` fails.

Reply:
The only way for `setOptimalLowPowerPaConfig` to fail at this point is SPI transfer error, since the value of `power` was already checked. If one SPI transaction fails, there's no reason to assume that subsequet ones will succeed. and this information should probably get to the user as quickly as possible.